src/withHTMLSubtitles/subtitlesParser.js

```diff
@@ -1,52 +1,74 @@
+                endTime: (c.endTime * 1000) | 0,
+                text: c.text
+            };
+            cues.push(cue);
+            cuesByTime[cue.startTime] = cuesByTime[cue.startTime] || [];
+            cuesByTime[cue.endTime] = cuesByTime[cue.endTime] || [];
         };
-        cues.push(cue);
-        cuesByTime[cue.startTime] = cuesByTime[cue.startTime] || [];
-        cuesByTime[cue.endTime] = cuesByTime[cue.endTime] || [];
-    };
-    parser.parse(text);
-    parser.flush();
-    window.VTTCue = nativeVTTCue;
-    cuesByTime.times = Object.keys(cuesByTime)
-        .map(function(time) {
-            return parseInt(time, 10);
-        })
-        .sort(function(t1, t2) {
-            return t1 - t2;
-        });
-    for (var i = 0; i < cues.length; i++) {
-        cuesByTime[cues[i].startTime].push(cues[i]);
-        var startTimeIndex = binarySearchUpperBound(cuesByTime.times, cues[i].startTime);
-        for (var j = startTimeIndex + 1; j < cuesByTime.times.length; j++) {
-            if (cues[i].endTime <= cuesByTime.times[j]) {
-                break;
+
+        parser.onparsingerror = function(error) {
+            if (error.code === CRITICAL_ERROR_CODE) {
+                reject(error);
+            } else {
+                console.warn('Subtitles parsing error', error);
+                errors.push(error);
             }
+        };
 
-            cuesByTime[cuesByTime.times[j]].push(cues[i]);
-        }
-    }
+        parser.onflush = function() {
+            cuesByTime.times = Object.keys(cuesByTime)
+                .map(function(time) {
+                    return parseInt(time, 10);
+                })
+                .sort(function(t1, t2) {
+                    return t1 - t2;
+                });
+            for (var i = 0; i < cues.length; i++) {
+                cuesByTime[cues[i].startTime].push(cues[i]);
+                var startTimeIndex = binarySearchUpperBound(cuesByTime.times, cues[i].startTime);
+                for (var j = startTimeIndex + 1; j < cuesByTime.times.length; j++) {
+                    if (cues[i].endTime <= cuesByTime.times[j]) {
+                        break;
+                    }
 
-    for (var k = 0; k < cuesByTime.times.length; k++) {
-        cuesByTime[cuesByTime.times[k]].sort(function(c1, c2) {
-            return c1.startTime - c2.startTime ||
-                c1.endTime - c2.endTime;
-        });
-    }
+                    cuesByTime[cuesByTime.times[j]].push(cues[i]);
+                }
+            }
+
+            for (var k = 0; k < cuesByTime.times.length; k++) {
+                cuesByTime[cuesByTime.times[k]].sort(function(c1, c2) {
+                    return c1.startTime - c2.startTime ||
+                        c1.endTime - c2.endTime;
+                });
+            }
+
+            // we may have multiple parsing errors here, but will only respond with the first
+            // if subtitle cues are available, we will not reject the promise
```

Comment:
how is that comment not obvious from the code?
if you think its needed i am fine with it